go.mod

```diff
@@ -1,5 +1,5 @@
 module github.com/Eyevinn/hls-m3u8
 
-go 1.16
+go 1.22
```

Comment:
Why do we need to upgrade to 1.22?
I think we should keep the minimum version required. Do we have a reason to go to 1.22?